librz/type/parser/c_cpp_parser.c

```diff
@@ -372,7 +372,9 @@ RZ_API RZ_OWN RzType *rz_type_parse_string_single(RzTypeParser *parser, const ch
 	ts_tree_delete(tree);
 	ts_parser_delete(tsparser);
 	free(patched_code);
-	return tpair ? tpair->type : NULL;
+	RzType *ret = tpair ? tpair->type : NULL;
+	free(tpair);
```

Comment:
Is it enough to use `free` on tpair? What about `tpair->base`? Is that still allocated?

Reply:
tpair->btype is stored in a hashtable.

Reply:
Yes, in general `RzType` are allocated and owned by the user, `RzBaseType` are stored in the hashtable and only borrowed (except edit/remove cases).

Reply:
Ok!